ДЗ

**Задача 1:**

*Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз*

a) в колоде 52 карты, из них одной масти 52/4=13 шт. Для формулы
$\displaystyle P(A)=\frac{m}{n}$ где m-число благоприятных исходов, n-общее число всех исходов.  
у нас будет  $\displaystyle n=C^4_{52}$-т.е. количество способов извлечь 4 карты из 52, а $\displaystyle m=C^4_{13}$, способов извлечь 4 карты одной масти.  
n и m рассчитываем по формуле:
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$  

In [1]:
from math import factorial
n = int(factorial(52) / (factorial(4) * factorial(52 - 4))) 
print (f'количество способов извлечь 4 карты из 52 равно {n}')
m = int(factorial(13) / (factorial(4) * factorial(13 - 4))) 
print (f'количество способов извлечь 4 карты одной масти равно {m}')
p = m/n
print (f'вероятность {p}')

количество способов извлечь 4 карты из 52 равно 270725
количество способов извлечь 4 карты одной масти равно 715
вероятность 0.0026410564225690276


б) 
Используем то, что сумма вероятностей противоположных событий равна $1$, т.е.  
$\displaystyle P(A)+P(\bar{A})=1$,  
где $\bar{A}$-событие, когда из 4-х карт не оказалось ни одного туза. 
 

Для этого события общее число исходов $\displaystyle n=C^4_{52}$,  а $\displaystyle m2=C^4_{48}$ - т.е. количество способов извлечь 4 карты из колоды без тузов (52-4туза=48). Тогда $\displaystyle P(A)=1-P(\bar{A})$  


$\displaystyle P(\bar{A})=\frac{m2}{n}$ где m2 - число исходов без тузов


𝑃(𝐴) обозначим как p2, n считали выше, рассчитаем m2 -

In [2]:
print (f'количество способов извлечь 4 карты из 52 равно {n}')
m2 = int(factorial(48) / (factorial(4) * factorial(48 - 4))) 
print (f'количество способов извлечь 4 карты одной масти равно {m2}')

количество способов извлечь 4 карты из 52 равно 270725
количество способов извлечь 4 карты одной масти равно 194580


In [3]:
p2 = 1 - m2/n
print (f'вероятность {p2}')


вероятность 0.2812632745405854


**ОТВЕТ**

а) вероятность того, что все карты – крести $\approx 0.0026 \approx 0.26\%$

б) вероятность, что среди 4-х карт окажется хотя бы один туз $\approx 0.2813 \approx 28.13\%$. 

**Задача 2:**

*На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?*


для формулы вероятности $\displaystyle P(A)=\frac{m}{n}$ общее число исходов будет определяться количеством сочетаний 3-х кнопок из 10, $\displaystyle n=C^3_{10}$, а количество благоприятных исходов m -замок открылся будет равен 1.

In [4]:
n = int(factorial(10) / (factorial(3) * factorial(10 - 3))) 
p = 1/n
print (f'вероятность {p}')

вероятность 0.008333333333333333


**ОТВЕТ**

вероятность открыть дверь с первой попытки $\approx 0.008333 \approx 0.83\%$

**Задача 3:**

*В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?*

Посчитаем по-другому)

Вероятность что первая вытащенная деталь окрашена $\displaystyle \frac{9}{15}$

вторая(при условии что 1я окрашена) $\displaystyle \frac{8}{14}$

третья(при условии что 1я и 2я окрашены) $\displaystyle \frac{7}{13}$

Общая вероятность тогда  $\displaystyle P(\text{все 3 из 3-х вытащенных деталей окрашены})=\frac{9}{15}\cdot\frac{8}{14}\cdot\frac{7}{13}$

In [5]:
print(f'P = {round(9/15*8/14*7/13,4)}')

P = 0.1846


In [12]:
print(f'для проверки посчитаем традиционнно, P = {(int(factorial(9) / (factorial(3) * factorial(9 - 3))))/(int(factorial(15) / (factorial(3) * factorial(15 - 3))))}')

для проверки посчитаем традиционнно, P = 0.18461538461538463


**ОТВЕТ**

вероятность того, что все извлеченные детали окрашены $\approx 0.1846 \approx 18.5\%$

**Задача 4:**

*В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?*

$\displaystyle P(A)=\frac{m}{n}$ где m-число благоприятных исходов, n-общее число всех исходов.  
у нас будет  $\displaystyle n=C^2_{100}$-т.е. всего вариантов купить 2 билета, а количество благоприятных исхтодов (выигр. 2 из 2-х) m=1.  
n рассчитываем по формуле:
$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$  

In [7]:
n = int(factorial(100) / (factorial(2) * factorial(100 - 2))) 
print (f'вариантов купить 2 из 100 равно {n}')
m = 1
p = m/n
print (f'вероятность {p}')


вариантов купить 2 из 100 равно 4950
вероятность 0.00020202020202020202
